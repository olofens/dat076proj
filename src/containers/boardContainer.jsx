import React from 'react';
import ReactDOM from 'react-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from "redux";
import { startEditTask, deleteTask, updateTime } from "../actions/index.js"
import BoardComponent from "../components/boardComponent.jsx"
import Droppable from "./droppable.jsx"

class BoardContainer extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <BoardComponent
                tasks={this.props.tasks}
                startEditTask={this.props.startEditTask}
                deleteTask={this.props.deleteTask}
                updateTime={this.props.updateTime}
            />
        );
    }
}

function mapStateToProps(state) {
    return {
        tasks: state.doingTasks
    }
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators({
        updateTime: updateTime,
        startEditTask: startEditTask,
        deleteTask: deleteTask
    }, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(BoardContainer);
