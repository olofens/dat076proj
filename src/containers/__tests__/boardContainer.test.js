import React from 'react';
import { shallow } from 'enzyme';
import boardContainer from '../boardContainer.jsx';


describe('boardContainer', () => {
  it('should render correctly in "debug" mode', () => {

    const component = shallow(<boardContainer debug />);
  
    expect(component).toMatchSnapshot();
  });
});