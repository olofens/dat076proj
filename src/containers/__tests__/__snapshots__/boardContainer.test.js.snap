// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`boardContainer should render correctly in "debug" mode 1`] = `
<boardContainer
  debug={true}
/>
`;
